packages/foam-vscode/src/features/filtered-resources.ts

```diff
@@ -0,0 +1,252 @@
+    );
+  }
+
+  refresh(): void {
+    this.computeFilteredResources();
+    this._onDidChangeTreeData.fire();
+  }
+
+  getTreeItem(item: FilteredResourceTreeItem): vscode.TreeItem {
+    return item;
+  }
+
+  getChildren(directory?: Directory): Thenable<FilteredResourceTreeItem[]> {
+    if (!directory && this.groupBy === FilteredResourcesConfigGroupBy.Folder) {
+      const directories = Object.entries(this.getFilteredResourcesByDirectory())
+        .sort(([a], [b]) => a.localeCompare(b))
+        .map(
+          ([dir, filteredResources]) =>
+            new Directory(dir, filteredResources, this.filteredNoteContextValue)
+        );
+      return Promise.resolve(directories);
+    }
+
+    if (directory) {
+      const filteredResources = directory.resources.map(
+        o => new FilteredResource(o, this.filteredNoteContextValue)
+      );
+      return Promise.resolve(filteredResources);
+    }
+
+    const filteredResources = this.filteredResources.map(
+      o => new FilteredResource(o, this.filteredNoteContextValue)
+    );
+    return Promise.resolve(filteredResources);
+  }
+
+  async resolveTreeItem(
+    item: FilteredResourceTreeItem
+  ): Promise<FilteredResourceTreeItem> {
+    if (item instanceof FilteredResource) {
+      const content = await this.dataStore.read(item.resource.uri);
+      item.tooltip = getNoteTooltip(content);
+    }
+    return item;
+  }
+
+  private computeFilteredResources(): void {
+    this.filteredResources = this.workspace
+      .list()
+      .filter(this.filterPredicate)
+      .filter(resource => !this.isMatch(resource.uri))
+      .sort(this.sort);
+  }
+
+  private isMatch(uri: URI) {
+    return micromatch.isMatch(uri.fsPath, this.exclude);
+  }
+
+  private getGlobs(fsPaths: string[], globs: string[]): string[] {
+    globs = globs.map(glob => (glob.startsWith('/') ? glob.slice(1) : glob));
+
+    const exclude: string[] = [];
+
+    for (const fsPath of fsPaths) {
+      let folder = fsPath.replace(/\\/g, '/');
+      if (folder.substr(-1) === '/') {
+        folder = folder.slice(0, -1);
+      }
+      exclude.push(...globs.map(g => `${folder}/${g}`));
+    }
+
+    return exclude;
+  }
+
+  private getFilteredResourcesByDirectory(): FilteredResourcesByDirectory {
+    const filtered: FilteredResourcesByDirectory = {};
+    for (const resource of this.filteredResources) {
+      const p = resource.uri.path.replace(this.root, '');
+      const { dir } = path.parse(p);
+
+      if (filtered[dir]) {
+        filtered[dir].push(resource);
+      } else {
+        filtered[dir] = [resource];
+      }
+    }
+
+    for (const k in filtered) {
+      filtered[k].sort(this.sort);
+    }
+
+    return filtered;
+  }
+
+  private sort(a: Resource, b: Resource) {
+    const titleA = getTitle(a);
+    const titleB = getTitle(b);
+    return titleA.localeCompare(titleB);
+  }
+}
+
+export interface FilteredResourcesProviderConfig
+  extends FilteredResourcesConfig {
+  workspacesFsPaths: string[];
```

Comment:
can we use `URI` here instead?
The goal is to reduce the exposure of platform related strings, until the very last place where they need to be used.
(e.g. here wherever you use `workspacePaths` you'd be using `workspaceFolders.fsPath` internally, and exposing just the URI to the client)